src/coreclr/jit/lsraxarch.cpp

```diff
@@ -2701,4 +2739,25 @@ void LinearScan::SetContainsAVXFlags(unsigned sizeOfSIMDVector /* = 0*/)
     }
 }
 
+inline regMaskTP LinearScan::BuildEvexIncompatibleMask(GenTree* tree)
```

Comment:
can we have `assert(!tree->isEvexCompatibleHWIntrinsic())`?

Reply:
We can't, because it's not always an intrinsic. For example, `GT_CNS_VEC` cannot use EVEX form as it is currently lowered, but it is not a HWIntrinsic.